jhipster-framework/src/main/java/io/github/jhipster/security/PersistentTokenCache.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016-2018 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jhipster.security;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+/**
+ * Simple time-limited cache for login tokens, necessary to avoid concurrent
+ * requests invalidating one another. It uses a {@link java.util.LinkedHashMap}
+ * to keep the tokens in order of expiration. During access any entries which
+ * have expired are automatically purged.
+ */
+public class PersistentTokenCache<T> {
+
+    private final long expireMillis;
+
+    private final LinkedHashMap<String, Value> map;
```

Comment:
Can't it be `private final Map<String, Value> map;`?

Reply:
Sure, with an extra `import` (or asterisk). Would that be preferable? The fact that it's `Linked` is relevant to this implementation, so I thought I would make that explicit also in the field type.

Reply:
It's your call Eric, I just prefer not to have a concrete implementation when declaring fields.

Reply:
I've always been taught you should use the interface here, so let's do a Map, at least for consistency with the rest of the code. Nobody will see this anyway as it's private, so it's no big deal.